src/modlist.cpp

```diff
@@ -638,17 +637,6 @@ bool ModList::setData(const QModelIndex &index, const QVariant &value, int role)
 
   emit postDataChanged();
 
-  IModList::ModStates newState = state(modID);
-  if (oldState != newState) {
-    try {
-      m_ModStateChanged(info->name(), newState);
-    } catch (const std::exception &e) {
-      log::error("failed to invoke state changed notification: {}", e.what());
-    } catch (...) {
-      log::error("failed to invoke state changed notification: unknown exception");
-    }
-  }
-
```

Comment:
As far as I can tell, in `setData`, a change of state can only happen if the mod was checked/unchecked, and since this part already calls `Profile::setModEnabled`, the plugin should be notified there.